### PR TITLE
Fixed broken sorting for SM/DS type in mgmt API

### PR DESCRIPTION
**Pull request by kaizimmerm**
* Sorting for the two types did not work at all…blame on me it seems I messed that up somewhere last year :smile: 
* SM type field MAXASSIGNMENTS was in filter/sort actually called differently ("MAX"). I fixed that also as sort filter params should always have the same name as the field in the mgmt. API payloads. I mean how should an API use ever figure this out if we use different names.

### CODE REVIEW

**Comment by michahirsch** [comment]
code seems ok but the circle CI build failed because of test-failure.

**Comment by kaizimmerm** [comment]
Circle is green?

File: hawkbit-rest-resource/src/main/java/org/eclipse/hawkbit/rest/resource/PagingUtility.java [hunk 1]
```diff
@@ -69,6 +71,17 @@ static Sort sanitizeSoftwareModuleSortParam(final String sortParam) {
         return sorting;
     }
 
+    static Sort sanitizeSoftwareModuleTypeSortParam(final String sortParam) {
+        final Sort sorting;
+        if (sortParam != null) {
```
Comment (SirWayne) [inline hawkbit-rest-resource/src/main/java/org/eclipse/hawkbit/rest/resource/PagingUtility.java]: if you early return you need no local variable and no else

   if (sortParam == null) {
         return new Sort(Direction.ASC, SoftwareModuleTypeFields.NAME.getFieldName());
      } 
        return  new Sort(SortUtility.parse(SoftwareModuleTypeFields.class, sortParam));

File: hawkbit-rest-resource/src/main/java/org/eclipse/hawkbit/rest/resource/PagingUtility.java [hunk 2]
```diff
@@ -80,6 +93,17 @@ static Sort sanitizeDistributionSetSortParam(final String sortParam) {
         return sorting;
     }
 
+    static Sort sanitizeDistributionSetTypeSortParam(final String sortParam) {
+        final Sort sorting;
+        if (sortParam != null) {
```
Comment (SirWayne) [inline hawkbit-rest-resource/src/main/java/org/eclipse/hawkbit/rest/resource/PagingUtility.java]: see above

**Comment by SirWayne** [comment]
+1

**Pull request merged**